monad-peer-discovery/src/discovery.rs

```diff
@@ -608,6 +615,123 @@ impl<ST: CertificateSignatureRecoverable> PeerDiscovery<ST> {
     fn is_pinned_node(&self, peer_id: &NodeId<CertificateSignaturePubKey<ST>>) -> bool {
         self.check_validator_membership(peer_id) || self.pinned_full_nodes.contains(peer_id)
     }
+
+    fn write_peers_to_file(&self) {
+        // store name records of all peers in routing_info and pending_queue
+        let mut peer_configs: Vec<NodeBootstrapPeerConfig<ST>> = self
+            .routing_info
+            .iter()
+            .map(|(peer_id, name_record)| NodeBootstrapPeerConfig {
+                address: name_record.udp_address().to_string(),
+                record_seq_num: name_record.seq(),
+                secp256k1_pubkey: peer_id.pubkey(),
+                name_record_sig: name_record.signature,
+                auth_port: name_record.name_record.authenticated_udp_port(),
+            })
```

Comment:
i would add try_from/try_into (or non-fallible variants) between name record and bootstrap peer config